Add move operator (#87)
open-telemetry/opentelemetry-log-collection: docs/operators/move.md

| `output`   | Next in pipeline | The connected operator(s) that will receive all outbound entries                                                                                                                                                                         |
| `from`      | required       | The [field](/docs/types/field.md)  to move the value out of.   
| `to`      | required       | The [field](/docs/types/field.md)  to move the value into.
| `on_error` | `send`           | The behavior of the operator if it encounters an error. See [on_error](/docs/types/on_error.md)                                                                                                                                          |
| `if`       |                  | An [expression](/docs/types/expression.md) that, when set, will be evaluated to determine whether this operator should be used for the given entry. This allows you to do easy conditional parsing without branching logic with routers. |

Example usage:

> renaming body value

Reviewer: ```suggestion
Rename a value
```

```yaml
- type: move
    from: key1
    to: key3
```

<table>
<tr><td> Input Entry</td> <td> Output Entry </td></tr>
<tr>
<td>

```json
{
  "resource": { },
  "attributes": { },  
  "body": {
    "key1": "val1",
    "key2": "val2"
  }
}
```

</td>
<td>

```json
{
  "resource": { },
  "attributes": { },  
  "body": {
    "key3": "val1",
    "key2": "val2"
  }
}
```

</td>
</tr>
</table>
<hr>

> Move a value called "uuid" from body to resource

Reviewer: ```suggestion
Move a value from the body to resource
```

```yaml
- type: move
    from: uuid
    to: $resoruce.uuid
```

<table>
<tr><td> Input Entry</td> <td> Output Entry </td></tr>
<tr>
<td>

```json
{
  "resource": { },
  "attributes": { },  
  "body": {
    "uuid": "091edc50-d91a-460d-83cd-089a62937738"
  }
}
```

</td>
<td>

```json
{
  "resource": { 
    "uuid": "091edc50-d91a-460d-83cd-089a62937738"
  },
  "attributes": { },  
  "body": { }
}
```

</td>
</tr>
</table>

<hr>

> Move a value called "ip" from body to attributes

Reviewer: ```suggestion
Move a value from the body to attributes
```

```yaml
- type: move
    from: ip
    to: $attributes.ip
```

<table>
<tr><td> Input Entry</td> <td> Output Entry </td></tr>
<tr>
<td>

```json
{
  "resource": { },
  "attributes": { },  
  "body": {
    "ip": "8.8.8.8"
  }
}
```

</td>
<td>

```json
{
  "resource": { },
  "attributes": { 
    "ip": "8.8.8.8"
  },  
  "body": { }
}
```

</td>
</tr>
</table>

<hr>

> Move a string from 'log' in nested in body to $body

Reviewer: ```suggestion
Replace the body with an individual value nested within the body
```

```yaml
- type: move
    from: nested.log
    to: $body
```

<table>
<tr><td> Input Entry</td> <td> Output Entry </td></tr>
<tr>
<td>

```json
{
  "resource": { },
  "attributes": { },  
  "body": {
    "log": "The log line"
  }
}
```

</td>
<td>

```json
{
  "resource": { },
  "attributes": { },  
  "body": "The log line"
}
```

</td>
</tr>
</table>

<hr>

Move an object from 'log' in body to $body
```yaml
- type: move
    from: log
    to: $body
```

<table>
<tr><td> Input Entry</td> <td> Output Entry </td></tr>
<tr>
<td>

```json
{
  "resource": { },
  "attributes": { },  
  "body": {
    "log": "The log line"
  }
}
```

</td>
<td>

```json
{
  "resource": { },
  "attributes": { },  
  "body": "The log line"
}
```

</td>
</tr>

> </table>

Reviewer: I believe this case is here by mistake and should be removed

<hr>

> Move an object from somewhere in nested in body to $body

Reviewer: ```suggestion
Remove a layer from the body
```

### Configuration Fields

| Field      | Default          | Description                                                                                                                                                                                                                              |
| ---        | ---              | ---                                                                                                                                                                                                                                      |
| `id`       | `restructure`    | A unique identifier for the operator                                                                                                                                                                                                     |
| `output`   | Next in pipeline | The connected operator(s) that will receive all outbound entries                                                                                                                                                                         |
| `from`      | required       | The field to move the value out of.   

> | `to`      | required       | The field to move the value into.

Reviewer: Can we link to the doc for `field`, so people can easily find out what syntax is expected here?

| `on_error` | `send`           | The behavior of the operator if it encounters an error. See [on_error](/docs/types/on_error.md)                                                                                                                                          |
| `if`       |                  | An [expression](/docs/types/expression.md) that, when set, will be evaluated to determine whether this operator should be used for the given entry. This allows you to do easy conditional parsing without branching logic with routers. |

Example usage:
```yaml
- type: move
    from: key1
    to: key3
```

<table>
<tr><td> Input Body </td> <td> Output Body </td></tr>
<tr>
<td>

```json
{
  "key1": "val1",
  "key2": "val2"
}
```

</td>
<td>

```json
{
  "key3": "val1",
  "key2": "val2"
}

> ```

Reviewer: Since we're supporting `resource`, `attributes`, and `body` now, this section seems too specific to movements within the `body`. Can we add a few more examples?

Let's have the table show `Input Entry` and `Output Entry`, and use the following format to represent the entries:
```json
{
  "resource": { },
  "attributes": { },  
  "body": {
    "key1": "val1",
    "key2": "val2"
  }
}
```

Let's have the following examples:
1. Move a value called "uuid" from body to resource
```json
{
  "resource": { },
  "attributes": { },  
  "body": {
    "uuid": "091edc50-d91a-460d-83cd-089a62937738"
  }
}
```
becomes
```json
{
  "resource": { 
    "uuid": "091edc50-d91a-460d-83cd-089a62937738"
  },
  "attributes": { },  
  "body": { }
}
```

2. Move a value called "ip" from body to attributes
```json
{
  "resource": { },
  "attributes": { },  
  "body": {
    "ip": "8.8.8.8"
  }
}
```
becomes
```json
{
  "resource": { },
  "attributes": { 
    "ip": "8.8.8.8"
  },  
  "body": { }
}
```

3. Rename a value in body (current example is fine)

4. Move a string from somewhere in nested in body to `$body` 
```json
{
  "resource": { },
  "attributes": { },  
  "body": {
    "log": "The log line"
  }
}
```
becomes
```json
{
  "resource": { },
  "attributes": { },  
  "body": "The log line"
}
```

5. Move an object from somewhere in nested in body to `$body`
```json
{
  "resource": { },
  "attributes": { },  
  "body": {
    "wrapper": {
      "key1": "val1",
      "key2": "val2",
      "key3": "val3"
    }
  }
}
```
becomes
```json
{
  "resource": { },
  "attributes": { },  
  "body": {
    "key1": "val1",
    "key2": "val2",
    "key3": "val3"
  }
}
```
